DDDEastAnglia.Tests/Helpers/Email/SessionSubmissionEmailSender_Should.cs

```diff
@@ -0,0 +1,196 @@
+﻿using DDDEastAnglia.Helpers.Email;
+using DDDEastAnglia.Helpers.File;
+using DDDEastAnglia.Models;
+using NSubstitute;
+using NUnit.Framework;
+using System;
+using System.Net.Mail;
+
+namespace DDDEastAnglia.Tests.Helpers.Email
+{
+    [TestFixture]
+    public class SessionSubmissionEmailSender_Should
+    {
+        [Test]
+        public void ThrowAnExceptionWhenConstructed_WhenTheSuppliedEmailSenderIsNull()
+        {
+            var messageFactory = Substitute.For<IMessageFactory>();
+            var fileContentsProvider = Substitute.For<IFileContentsProvider>();
+            Assert.Throws<ArgumentNullException>(() => new SessionSubmissionEmailSender(null, messageFactory, fileContentsProvider));
+        }
+
+        [Test]
+        public void ThrowAnExceptionWhenConstructed_WhenTheSuppliedMessageFactoryIsNull()
+        {
+            var emailSender = Substitute.For<IEmailSender>();
+            var fileContentsProvider = Substitute.For<IFileContentsProvider>();
+            Assert.Throws<ArgumentNullException>(() => new SessionSubmissionEmailSender(emailSender, null, fileContentsProvider));
+        }
+
+        [Test]
+        public void ThrowAnExceptionWhenConstructed_WhenTheSuppliedFileContentsProviderIsNull()
+        {
+            var emailSender = Substitute.For<IEmailSender>();
+            var messageFactory = Substitute.For<IMessageFactory>();
+            Assert.Throws<ArgumentNullException>(() => new SessionSubmissionEmailSender(emailSender, messageFactory, null));
+        }
+
+        [Test]
+        public void SendAnEmailFromTheCorrectEmailAddressAndName()
```

Comment:
These first two tests should, I think, be common to all emails we send from the app, and should be guaranteed by the `IEmailSender` rather than the `SessionSubmissionEmailSender`. I'm in two minds about the next two tests, but I think they should be a level lower too.